### DISCUSSION TITLE
Barcode Image Provider using ZXing.Net

### DISCUSSION THREAD

**Discussion by pri-kise** [microsoft/ALAppExtensions#20136] [General]
Right now we are using the [ZXing.Net Library](https://github.com/micjahn/ZXing.Net) to generate Barcode Images for our .Net Customers.

Since want to set our own Apps to Target Cloud, we cannot use .Net in our Apps anymore.

Therefore I'd like to know if Microsoft would accept a PullRequest to the System Application using this OpenSource Library to generate Barcode Images. I know this DLL would have to provided by the Plattform, that we could use this non

I took some time to study the current Interfaces for providing a new Barcode Provider and Encoder.
ZXing.Net
https://github.com/microsoft/ALAppExtensions/tree/main/Modules/System/Barcode

My draft concept would be to add a new provider and encoder Interface for 1D Barcode Image Provider and Encoder.
Then I could provide an implementation for 1D Barcode Image using the ZXing.Net library (Assemblies are available for .Net Core)

**Comment by JesperSchulz** [comment]
Let me ask our platform team if this is something we can add to the BC platform.

**Reply by JesperSchulz** [reply]
I ran into some blockers, which will prevent us from adding ZXing. There are some issues with licensing (Apache License, not MIT). I also received a question, why the implemented fonts cannot be used? The received suggestion is to use Azure functions, if this really is needed. Sorry, this one seems to be a no-go for now.

**Reply by pri-kise** [reply]
Currently we still have customers OnPrem who don't want to move to SaaS within the next 5 years. And we can't explain our customer why the have to pay for a Barcode Font after Updating to BC while they where able to use the XZing Library for Free.

**Reply by JesperSchulz** [reply]
I see... I'll keep asking around. Maybe a door opens somewhere else.
